src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy

```diff
@@ -137,42 +149,47 @@ class NUnit extends ConventionTask {
     }
 
     def decideExecutionPath(Closure singleRunAction, Closure multipleRunsAction) {
-        if (!parallelForks || !test) {
-            return singleRunAction(test)
+        def input = getRunActionInput();
+
+        if (!parallelForks || !input) {
+            return singleRunAction(input)
         } else {
-            return multipleRunsAction(test)
+            return multipleRunsAction(input)
         }
     }
 
-    def singleRunExecute(def test) {
-        def testRuns = getTestInputsAsString(test)
-        testRun(testRuns, getTestReportPath())
+    def singleRunExecute(def input) {
+        def runs = getTestInputsAsString(input)
+        run(runs, getTestReportPath())
     }
 
-    def multipleRunsExecute(def test) {
-        def intermediatReportsPath = new File(getReportFolderImpl(), "intermediate-results-" + name)
-        intermediatReportsPath.mkdirs()
+    def multipleRunsExecute(def input) {
+        def intermediateReportsPath = new File(getReportFolderImpl(), "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
 
-        def testRuns = getTestInputAsList(test)
+        def runs = getTestInputAsList(input)
         GParsPool.withPool {
-            testRuns.eachParallel { testRun(it, new File(intermediatReportsPath, it + ".xml")) }
+            runs.eachParallel {
+                def fileName = toFileName(it)
```

Comment:
The intermediary filename for each parallel run done on v3 will
be a random UUID since 'where' conditions don't translate well to file
names. This is in contrast to the existing v2 style where a _namespace_
that is the argument for a **test** condition does make for a decent filename